### PR TITLE
Add support for callbacks

### PR DESCRIPTION
**Pull request by Deradon**
> You can hook into some of the dashboard actions to run custom business logic.

## Syntax

```
SplitWimduDashboard.configure do |config|
  config.observers.add(topic: :finish_experiment) do |experiment|
    # do something with `experiment`
  end
end
```

## Supported callbacks

* `start_experiment`
* `finish_experiment`
* `reset_experiment`
* `reopen_experiment`
* `delete_experiment`

### CODE REVIEW

File: spec/split_wimdu_dashboard/observers_spec.rb [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    end
+  end
+
+  describe "any?" do
+    subject { observers.any? }
+
+    context "when an observer is registered" do
+      before { observers.add(topic: :foo, &spy) }
+
+      it { expect(observers.any?).to be(true) }
```
Comment (Deradon) [inline spec/split_wimdu_dashboard/observers_spec.rb]: => `it { is_expected.to be(true) }`

File: lib/split_wimdu_dashboard/observers.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    def notify(topic:, args: [])
+      observers[topic].each { |callback| callback.call(*args) }
+    end
+
+    def reset
+      @observers = nil
+    end
+
+    def any?
+      !observers.empty?
```
Comment (tomasz-buchta) [inline lib/split_wimdu_dashboard/observers.rb]: you may use `observers.any?`

**Comment by lukaszklis** [comment]
@Deradon oh hai! :wave: :D

**Comment by paranoida** [comment]
I like it

**Comment by AlexParamonov** [comment]
looks good!

**Comment by Deradon** [comment]
@lukaszklis @paranoida @AlexParamonov 

Looks like it's been approved. Will merge it tomorrow xD

**Pull request merged**